terraform/ec2/linux/main.tf

```diff
@@ -26,6 +26,51 @@ locals {
   private_key_content = var.ssh_key_name != "" ? var.ssh_key_value : tls_private_key.ssh_key[0].private_key_pem
 }
 
+#####################################################################
+# Create EFS
+#####################################################################
+resource "aws_efs_file_system" "efs" {
+  creation_token = "efs-${random_id.testing_id.hex}"
+  tags           = {
+    Name = "efs-${random_id.testing_id.hex}"
+  }
+}
+
+resource "aws_efs_mount_target" "mount" {
+  file_system_id = aws_efs_file_system.efs.id
+  subnet_id = aws_instance.cwagent.subnet_id
+  security_groups = [aws_security_group.ec2_security_group.id]
+}
+
+resource "null_resource" "mount_efs" {
+  depends_on = [
+    aws_efs_mount_target.mount,
+    aws_instance.cwagent
+  ]
+
+  connection {
+    type = "ssh"
+    user = "ec2-user"
+    private_key = local.private_key_content
+    host = aws_instance.cwagent.public_ip
+  }
+
+  provisioner "file" {
+    source = "install-efs-utils.sh"
+    destination = "/tmp/install-efs-utils.sh"
+  }
+
+  provisioner "remote-exec" {
+    # https://docs.aws.amazon.com/efs/latest/ug/mounting-fs-mount-helper-ec2-linux.html
+    inline = [
+      "chmod +x /tmp/install-efs-utils.sh",
+      "/tmp/install-efs-utils.sh",
+      "sudo mkdir ${var.efs_mount_point}",
```

Comment:
Will the `efs_mount_point` change according to OS? If not, then IMO we don't need the variable though since its a static mount or partition.

Reply:
No, it will not change. I only made it a variable for code reuse

Reply:
Then it safe to delete it though IMO. Make it static and let's people less worry about it would be better experience.

Reply:
To help my learning, is that a design or stylistic principle?

Reply:
It's design though. I will leave the final decision to others though.